### PR TITLE
Flatten data check action ``validate`` API

### LINKED ISSUE #3243: Update data check's ``validate()`` method to return a flat hierarchy of messages
Per discussion with team, perhaps we should remove the dictionary output and instead, update the data check's ``validate()`` method to return a flat hierarchy of messages.

(Essentially, reverting https://github.com/alteryx/evalml/issues/1325)

### PR DESCRIPTION
**Pull request by angela97lin**
Closes #3243. This is being merged into my DC branch, not main! 

Lots of lines touched but not much logic.

### CODE REVIEW

File: docs/source/release_notes.rst [hunk 1]
```diff
@@ -25,6 +26,8 @@
     **Breaking Changes**
         * Renamed ``DateTime Featurizer Component`` to ``DateTime Featurizer`` and ``Natural Language Featurization Component`` to ``Natural Language Featurizer`` :pr:`3192`
         * Renamed ``HighlyNullDataCheck`` to ``NullDataCheck`` :pr:`3197`
+        * Flattened data check ``validate()`` output. See documentation for examples. :pr:`3244`
```
[the remark this answers is not in the document]
Comment (eccabay) [reply docs/source/release_notes.rst:29]: Also, the changelog is part of the documentation. Could you mention a specific page for users reading the changelog to refer to for examples?

**Review by eccabay — COMMENTED** [review]
I'm a big fan of this simplification!

**Review by freddyaboulton — APPROVED** [review]
@angela97lin I agree the new api format is easier to understand by  treating actions/warnings/errors the same. If the team is ok with the breaking change then this looks good to me!

**Review by ParthivNaresh — APPROVED** [review]
Awesome I love it!

File: docs/source/release_notes.rst (hunk 1 above)
[the remark this answers is not in the document]
Comment (angela97lin) [reply docs/source/release_notes.rst]: For sure, good points! :)

**Pull request merged**